### ISSUE TITLE
having no map options, i.e. empty dict, leads to nextflow error

### ISSUE THREAD

**Issue by sergpolly** [mirnylab/distiller-nf#142]
we commented the extra mapping options in the `project.yml`:
https://github.com/mirnylab/distiller-nf/blob/5b5da693f3835e2ed57635fbda13ef0d170c4f29/test/test_project.yml#L99

and the whole thing crashed - nextflow/groovy level crash that was:
```sh
N E X T F L O W  ~  version 19.01.0
Launching `distiller.nf` [berserk_kowalevski] - revision: c454203cd8
ERROR ~ Cannot invoke method get() on null object

 -- Check script 'distiller.nf' at line: 110 or see '.nextflow.log' file for more details
```

we should check into how are we using `map` dictionary and prevent crashes like that ...

**Comment by golobor** [comment]
yes, this and other issues are caused by the fact that we do not validate
project.yaml's.
Since there are so many different parameters to check, we would need some
kind of generic approach.
There is an ongoing discussion on how to implement such a feature in core
nextflow: https://github.com/nextflow-io/nextflow/issues/866 , but it seems
to have stuck.
Should we just implement something of our own?
